detection/distance_detector.py

```diff
@@ -0,0 +1,112 @@
+import argparse
+
+from utils.datasets import *
+from utils.utils import *
+
+import os
+from pathlib import Path
+import copy
+
+
+BASE_PATH = Path(os.path.dirname(os.path.abspath(__file__)))
+IMAGE_SIZE = 640
+CONF_TRESHOLD = 0.3
+IOU_THESHOLD = 0.5
+
+class SocialDistancing:
+    
+    def __init__(self, path='weights/yolov5s.pt') -> None:
+        self.device = torch_utils.select_device()
+        self.model = torch.load(BASE_PATH / path, map_location=self.device)['model'].float().fuse().eval()
+        self.names = self.model.names if hasattr(self.model, 'names') else self.model.modules.names
+
+    def __call__(self, image):
+        results = []
+
+        h,w,_  = image.shape
+        image_ = copy.deepcopy(image)
+
+        height_0, width_0 = image.shape[:2]  # orig hw
+        resized = IMAGE_SIZE / max(height_0, width_0)  # resize image to img_size
+
+        if resized != 1:  # always resize down, only resize up if training with augmentation
+            interp = cv2.INTER_AREA if resized < 1  else cv2.INTER_LINEAR
+            image_ = cv2.resize(image_, (int(width_0 * resized), int(height_0 * resized)), interpolation=interp)
+
+        image_size = check_img_size(IMAGE_SIZE)  # check img_size
+
+        image_ = letterbox(image, new_shape=image_size)[0]
+        
+        # Convert
+        image_ = image_[:, :, ::-1].transpose(2, 0, 1).copy()  # BGR to RGB, to 3x416x416
+
+        image_ = torch.from_numpy(image_).to(self.device)
+        image_ = image_.float()  # uint8 to fp16/32
+        image_ /= 255.0  # 0 - 255 to 0.0 - 1.0
+        if image_.ndimension() == 3:
+            image_ = image_.unsqueeze(0)
+
+        # Inference
+        predictions = self.model(image_, augment=True)[0]
+
+        # Apply NMS
+        predictions = non_max_suppression(predictions, CONF_TRESHOLD, IOU_THESHOLD, fast=True, agnostic=True)
+
+        for _, detections in enumerate(predictions):  # detections per image
+            if detections is not None and len(detections):
+                # Rescale boxes from img_size to im0 size
+                detections[:, :4] = scale_coords(image_.shape[2:], detections[:, :4], image.shape).round()
+
+                for *coors, conf, cls in detections:
+                    label = self.names[int(cls)]
+                    if label == 'person':
+                        results.append((label, coors))
+
+        return results
```

Comment:
the following code is being repeated in the three files, please make it such that it does not repeat.
1) detection/face_detector.py
2) detection/distance_detector.py
3) detection/mask_detector.py

Reply:
Reviews Done in [b333bd9](https://github.com/MuhammadAmir5670/intelligent-covid-SOP-compliance/pull/1/commits/b333bd972b0250753f031e29491cf0db2a105ce1) and [76b1304](https://github.com/MuhammadAmir5670/intelligent-covid-SOP-compliance/pull/1/commits/76b13040a4b36a906e298232437019c8a653987c)